cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -1282,6 +1292,251 @@ def _check_column_availability(self):
+                "'EFO:0010961' (Visium Spatial Gene Expression) and uns['spatial']['is_single'] is True."
+            )
+            # Exit as library_id is missing.
+            return
+
+        # Confirm shape of library_id is valid: allowed keys are images and scalefactors.
+        library_id_key = library_ids[0]
+        uns_library_id = uns_spatial[library_id_key]
+        if not self._has_no_extra_keys(uns_library_id, ["images", "scalefactors"]):
+            self.errors.append("uns['spatial'][library_id] can only contain the keys 'images' and 'scalefactors'.")
+
+        # images is required.
+        if "images" not in uns_library_id:
+            self.errors.append("uns['spatial'][library_id] must contain the key 'images'.")
+        # images is specified: proceed with validation of images.
+        else:
+            # Confirm shape of images is valid: allowed keys are fullres and hires.
+            uns_images = uns_library_id["images"]
+            if not self._has_no_extra_keys(uns_images, ["fullres", "hires"]):
+                self.errors.append(
+                    "uns['spatial'][library_id]['images'] can only contain the keys 'fullres' and 'hires'."
+                )
+
+            # hires is required.
+            if "hires" not in uns_images:
+                self.errors.append("uns['spatial'][library_id]['images'] must contain the key 'hires'.")
+            # hires is specified: proceed with validation of hires.
+            else:
+                self._validate_spatial_image_shape("hires", uns_images["hires"], 2000)
+
+            # fullres is optional.
+            uns_fullres = uns_images.get("fullres")
+            if uns_fullres is not None:
+                self._validate_spatial_image_shape("fullres", uns_fullres)
+
+        # scalefactors is required.
+        if "scalefactors" not in uns_library_id:
+            self.errors.append("uns['spatial'][library_id] must contain the key 'scalefactors'.")
+        # scalefactors is specified: proceed with validation of scalefactors.
+        else:
+            # Confirm shape of scalefactors is valid: allowed keys are spot_diameter_fullres and tissue_hires_scalef.
+            uns_scalefactors = uns_library_id["scalefactors"]
+            if not self._has_no_extra_keys(uns_scalefactors, ["spot_diameter_fullres", "tissue_hires_scalef"]):
+                self.errors.append(
+                    "uns['spatial'][library_id]['scalefactors'] can only contain the keys "
+                    "'spot_diameter_fullres' and 'tissue_hires_scalef'."
+                )
+
+            # spot_diameter_fullres is required.
+            if "spot_diameter_fullres" not in uns_scalefactors:
+                self.errors.append(
+                    "uns['spatial'][library_id]['scalefactors'] must contain the key 'spot_diameter_fullres'."
+                )
+            # spot_diameter_fullres is specified: proceed with validation.
+            else:
+                self._validate_float(
```

Comment:
nit: I think it'd be useful to include more details in the error message (i.e. "This must be the value of the spot_diameter_fullres field from scalefactors_json.json"), in which case we probably want to pull this out of a generalized `_validate_float` method